ntoskrnl/include/internal/se.h

```diff
@@ -71,6 +61,14 @@ typedef struct _TOKEN_AUDIT_POLICY_INFORMATION
 #define TOKEN_DUPLICATE_METHOD 0xDUL
 #define TOKEN_FILTER_METHOD    0xFUL
 
+//
+// Access Rights Types Array
+//
+#define ACCESS_RIGHTS_REMAINING_RIGHTS_ARRAY  0
+#define ACCESS_RIGHTS_GRANTED_RIGHTS_ARRAY    1
+#define ACCESS_RIGHTS_DENIED_RIGHTS_ARRAY     2
+#define ACCESS_RIGHTS_MAX_ARRAY               3
```

Comment:
I can understand why you would not want an allocation, but what was wrong with the struct?
Imo this is less sensitive to errors than an array?